java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/WeightActivity.java

```diff
@@ -99,21 +103,49 @@ private void putWeight(String value) {
 	}
 
 	private void renderWeights(List<Thing2> things) {
-		int count = 0;
+		int xPoint = 0;
+		GraphView graph = (GraphView) findViewById(R.id.graph);
+		LineGraphSeries<DataPoint> series = new LineGraphSeries<>();
+		int size = things.size();
+		DataPoint dataPointArray[] = new DataPoint[size];
+
+		// Populate the last weight view
+		Weight lastWeightValue  = (Weight)things.get(0).getData();
+		TextView lastWeightTextView = (TextView) findViewById(R.id.lasWeight);
+		lastWeightTextView.setText(String.valueOf(lastWeightValue.getValue().getKg()));
+
+		// Populate the list and last weight
+		mAdapter.clear();
 		for(Thing2 thing : things) {
-			Weight w = (Weight)thing.getData();
-		if (count < 1) {
-			TextView lastWeight = (TextView) findViewById(R.id.lasWeight);
-			lastWeight.setText(String.valueOf(w.getValue().getKg()));
-			count++;
+			Weight weight = (Weight) thing.getData();
+			mAdapter.add(String.valueOf(weight.getWhen().getDate().getM() + "/" +
+					weight.getWhen().getDate().getD() + "/" +
+					weight.getWhen().getDate().getY()) + "               " + "                    " +
+					String.valueOf(weight.getValue().getKg()));
+			mAdapter.notifyDataSetChanged();
 		}
 			final int month = w.getWhen().getDate().getM();
 			final int day = w.getWhen().getDate().getD();
 			final int year = w.getWhen().getDate().getY();
 			mAdapter.add(String.valueOf(String.format(month + "/" + day + "/" + year)
 					+ "                                  " + String.valueOf(w.getValue().getKg())));
 			mAdapter.notifyDataSetChanged();
+
+		// Populate the weight chart data points
+		Collections.reverse(things);
+		for(Thing2 weightThing : things) {
+			Weight chartWeight = (Weight) weightThing.getData();
+			DataPoint dataPoint = new DataPoint(xPoint, chartWeight.getValue().getKg().intValue());
+			dataPointArray[xPoint] = dataPoint;
+			xPoint++;
 		}
+
+		series.resetData(dataPointArray);
+		series.setColor(Color.WHITE);
+		graph.setTitle("lbs");
```

Comment:
in all the examples above we are calling getKg() but this is labelling the series at "lbs", seems like it should be "kg"